### PR TITLE
feat!: improve naming, fix module url

### PR DESCRIPTION
**Pull request by toddbaert**
- Fixes module location
- changes flagd root level schema name to `flags.json` instead of `flagd_definitions`
  - "flags" is the main top level object of the root schema, and though we have others (like `$evalutors` they are optional and used in `flags`.
  - it seems much more sensible as a public schema: https://flagd.dev/schema/v0/flagd-definitions.json vs https://flagd.dev/schema/v0/flags.json

Fixes: https://github.com/open-feature/flagd-schemas/issues/125

### CODE REVIEW

File: .github/workflows/release-please.yaml [hunk 1]
```diff
@@ -44,20 +44,3 @@ jobs:
...
-      # Run `git checkout`
-      - uses: actions/checkout@v3
-        with:
-          ref: ${{ needs.release-please.outputs.json_release_tag }}
-      - uses: JasonEtco/create-an-issue@v2
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          TAG: ${{ needs.release-please.outputs.json_release_tag }}
-        with:
-          filename: .github/schema-store-issue-template.md
```
Comment (beeme1mr) [inline .github/workflows/release-please.yaml:63]: Why did you remove this?

**Review by beeme1mr — COMMENTED** [review]
What's the impact of this change?

File: .github/workflows/release-please.yaml (hunk 1 above)
Comment (toddbaert) [reply .github/workflows/release-please.yaml:63]: Because it's currently broken and it's not longer something we want to connect directly to this repo (the schema is hosted at flagd.dev).

**Comment by toddbaert** [comment]
> What's the impact of this change?

The public URL will be more sensible, and a 1-line breaking change will need to be absorbed in flagd (where the schema is loaded).

File: json/flagd_definitions.go [hunk 2]
```diff
@@ -2,8 +2,8 @@ package flagd_definitions
...
 import _ "embed"
 
-//go:embed flagd-definitions.json
-var FlagdDefinitions string
+//go:embed flags.json
+var FlagSchema string
 
 //go:embed targeting.json
-var Targeting string
+var TargetingSchema string
```
Comment (Kavindu-Dodan) [inline json/flagd_definitions.go:9]: We should track and change flagd references - https://github.com/open-feature/flagd/blob/main/core/pkg/evaluator/json.go#L407-L414

**Review by Kavindu-Dodan — APPROVED** [review]
Left [this](https://github.com/open-feature/flagd-schemas/pull/128/files#r1456487635) comment but otherwise I think this is good change

File: json/flagd_definitions.go (hunk 2 above)
Comment (toddbaert) [reply json/flagd_definitions.go:9]: Yes I am going to immediately absorb this in flagd when this is merged, since I want the better schema name.

File: .github/schema-store-issue-template.md [hunk 3]
```diff
@@ -21,9 +20,14 @@ Extend the catalog entry found in `src/api/json/catalog.json` to include the new
...
     ],
-    "url": "https://raw.githubusercontent.com/open-feature/schemas/main/json/flagd-definitions.json",
+    "url": "https://flagd.dev/schema/v0/flags.json",
     "versions": {
     "0.1.1": "https://raw.githubusercontent.com/open-feature/schemas/json/json-schema-v0.1.1/json/flagd-definitions.json",
 +   "X.X.X": "https://raw.githubusercontent.com/open-feature/schemas/json/json-schema-vX.X.X/json/flagd-definitions.json"
 }
-```
\ No newline at end of file
+```
```
Comment (toddbaert) [inline .github/schema-store-issue-template.md:28]: Now this file suggests reviewing the schemastore version, and updating the version hosted at flagd.dev

File: .github/workflows/release-please.yaml (hunk 1 above)
Comment (toddbaert) [reply .github/workflows/release-please.yaml:63]: I've added this back with changes: https://github.com/open-feature/flagd-schemas/pull/128/files#r1457940078

**Review by beeme1mr — APPROVED** [review]
